odoo/models.py

```diff
@@ -1698,6 +1699,97 @@ def _read_group_fill_results(self, domain, groupby, remaining_groupbys,
+        user, so we change that. We add some fake records for each missing months
+        between the earliest and the lastest date of the result
+                                                     ___
+                                 ___                |   |
+                                |   |               |   |
+                                |   | ___           |   |
+                                |   ||   |          |   |
+                                |___||___| ___  ___ |___|
+                                 Aug  Sep  Oct  Nov  Dec
+
+        :param list data: the data containing groups
+        :param list groupby: name of the first group by
+        :param datetime start: start of the interval, if None, min date is used
+        :param datetime end: end of the interval, if None max date is used
+        :param relativedelta interval:
+        :rtype: list
+        :return: list
+        """
+        groupby_name = groupby[0]
+
+        # existing non null datetimes
+        existing = [d[groupby_name] for d in data if d[groupby_name]]
+
+        if len(existing) < 2:
+            return data
+
+        first, last = existing[0], existing[-1]
+
+        expected = list(date_utils.date_range(first, last, interval))
+
+        if len(existing) < len(expected):
+            tz = data[0][groupby_name].tzinfo
+            empty_data = dict.fromkeys(aggregated_fields, False)
+            empty_data['id'] = False
+            empty_data[groupby_name.split(':')[0] + '_count'] = 0
+            if not data[-1][groupby_name]:
+                nulls = [data[-1]]
+                data = collections.deque(data[:-1])
+            else:
+                nulls = []
+                data = collections.deque(data)
+
+            new_data = []
+
+            # Important Note: Here we have 'expected' which contains the list of
+            # the dates that should be represented inside data when it is returned.
+            # Note that 'expected' is sorted as well as data so what we do is empty
+            # data progressivly by popping its elements. To do this properly, we
+            # need to compare dates together but here comes a little problem :
+            # In some circumstances, dates can have a different timezone resulting
+            # in a false negative. To avoid that, we remove that timezone but it
+            # could be an error.
+
+            for dt in expected:
+                if isinstance(data[0][groupby_name].tzinfo, psycopg2.tz.FixedOffsetTimezone):
+                    # Here we assume that tz comes from PG after it date_trunced
+                    # a DATE (not a TIMESTAMP without TZ).
+                    data[0][groupby_name] = data[0][groupby_name].replace(tzinfo=None)
+
+                if data and data[0][groupby_name] == dt:
+                    new_data.append(data.popleft())
+                else:
+                    new_data.append(dict(empty_data, **{groupby_name: dt}))
```

Comment:
Can `groupby_name` contain `':'`?
If that is the case, is it correct to use the key `groupby_name`?